Automattic/wp-calypso
# Jetpack Settings: Introduce dev mode notice for settings

Some modules are unavailable in Jetpack when the site is in development mode, and this is by design. We're supposed to disable the cards that manage those modules if the site is in dev mode. This PR introduces a notice for development mode in Calypso settings, similar to the one from https://github.com/Automattic/jetpack/pull/6180.

![](https://cldup.com/m7l9XFGyFr.png)

This notice will be displayed if the site is a Jetpack site and in development mode by either constant, filter or enforced by the URL (localhost).

This PR will also go together with additional PRs that will handle disabling each of the modules if the site is in development mode:

* After the Deadline: #11779 
* Post by Email: #11781
* Photon: #11783
* Subscriptions: #11784

To test:
* Checkout this branch
* Enable dev mode by either using:
  * The `jetpack_development_mode` filter
  * The `JETPACK_DEV_DEBUG` constant
* Verify the notice is shown above the section navigation, as shown on the screenshot.
* Disable dev mode and refresh Calypso.
* Verify the notice no longer shows.
* Verify WordPress.com sites are not affected at all.
